### ISSUE TITLE
side effect of `focuslost`

### ISSUE THREAD

**Issue by Shane-XB-Qian** [vim/vim#7703]
**Describe the bug**
`focuslost` (with `system()`) looks suspended `vimresized` event.

**To Reproduce**
vi --clean -Nu <(cat<<EOF
au focuslost * silent call system('ls')
au vimresized * echomsg 'foo'
EOF
)
// no foo.

**Expected behavior**
// had foo.

**Environment (please complete the following information):**
 - v8.2.2374
 - xterm

**Additional context**
side effect of #7673
not sure if #7674 and #7675

**Comment by chrisbra** [comment]
> Expected behavior
> // had foo.

why would you expect this? columns and lines does not change, the vim window has not been resized. The Gui does also not output anything, so I think the current behavior is expected.

**Comment by Shane-XB-Qian** [comment]
> why would you expect this? columns and lines does not change, the vim window has not been resized

um...that needed some manual work to 'resize'... :-)



> The Gui

after #7673 , we are talking about Tui vim actually. (focuslost event in tui vim)

**Comment by chrisbra** [comment]
> um...that needed some manual work to 'resize'... :-)

There is no resized involved here.

> after #7673 , we are talking about Tui vim actually. (focuslost event in tui vim)

of course, but it matters to the gui at the same time and it should behave consistently in both TUI and the GUI.

**Comment by Shane-XB-Qian** [comment]
> > um...that needed some manual work to 'resize'... :-)
> 
> There is no resized involved here.

pls try to manually issue it.. sorry i didnot say very complete words... -_-#



> it matters to the gui at the same time and it should behave consistently in both TUI and the GUI.

maybe yes, but perhaps no, since in tui it was \033[I and [O, looks they are different..
and i actually never use gvim... :-)

**Comment by brammool** [comment]
I tried a few things, but do get the "foo" message in all cases.
Can you also reproduce it with ":set columns=88" somehow, to rule out that manually resizing the terminal happens
when the terminal is not in focus?
Perhaps the resize event is triggered while executing the FocusLost autocommand?  Then adding ++nested might help.
That is not a solution but might help understanding your problem.

**Comment by Shane-XB-Qian** [comment]
yes, it can get 'foo' if issued by vim ex, but mostly 'resize' looks was kicked off just by user manually changed the terminal size by window manger.
- in that case, it cannot.

and you're right, ++nested helped. however it did Not help on real xterm, but help e.g xfce-term.
- i saw you changed the judgement of such macro #ifdef recent, and actually \033[I and [O had a bit different behavior/output in different terminal. perhaps related.
- and as you said, ++nested looks not a good solution..

**Comment by Shane-XB-Qian** [comment]
by the way, actually there is another issue though not the same issue as this ticket.
e.g mouse cursor on terminal A (vim opened there), and you're working (the 'focus') on terminal B, then try to scroll (e.g by double fingers on touchpad), that/such 'scroll' actually would happen on terminal A, and vim at A would got 'focusgained' event, but (the problem is) looks a bit delay as (same time as) scrolling happened..
// if it's difficult to understand, then let it be, just try to tell you another flaw of 'focus'......
// solution perhaps was os level let 'focus' can stick on B unless switch 'foucs' by 'click' but not by 'mouse' cursor, or vim on A can get 'focusgained' event as soon as possible to prevent the scrolling.

**Comment by Shane-XB-Qian** [comment]
though not smooth, but mostly worked if `resize` not via vim cmd `:resize 123`, but terminal window manager itself..
// so closing.. let it be...

#10803
